### ISSUE TITLE
FCMRebooted error on iOS 17.2 with Delphi 12

### ISSUE THREAD

**Issue by elcharlie** [DelphiWorlds/Kastri#212]
I'm trying to run the FCMREbooted demo on iOS and it doesn't work. I have created an application from scratch, following the steps, and it still gives me the same error, I put them in case it helps:

[PAClient Error] Error: E0776 2024-02-16 09:36:05.834 xcodebuild[21138:999430] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path "/var/folders/q6/lhwpt4n1537fhr9ckp80h4kr0000gn/T/FCMRebooted_2024-02-16_09-36-05.833.xcdistributionlogs".
[PAClient Error] Error: E0776 error: exportArchive: No Team Found in Archive
[PAClient Error] Error: E0776 Error Domain=IDEDistributionAnalyzeArchiveStepErrorDomain Code=0 "No Team Found in Archive" UserInfo={NSLocalizedDescription=No Team Found in Archive, NSLocalizedRecoverySuggestion=Use the Signing & Capabilities editor to assign a team to the targets and build a new archive.}
[PAClient Error] Error: E0776 ** EXPORT FAILED **
Failed

![16-02-2024_09-49-48](https://github.com/DelphiWorlds/Kastri/assets/10128544/ea1ded09-d410-48cc-ab7f-b6e7284af17f)

When I run the Firebase Project Doctor utility, I get an error when checking the GoogleService-Info.plist, but it is added to the project. I don't know where the error is.
I also leave the log generated from Firebase Project Doctor:

Swift Compatibility check:
Folders check PASSED for: C:\Users\Administrador\Documents\Embarcadero\Studio\SDKs\iPhoneOS17.0.sdk

Linker Options check:
Found linker option for "Cfg_2" config - Value: -ObjC -rpath /usr/lib/swift
Linker option should be applied to "All configurations"

Google Services check:
Failed to parse GoogleService-Info.plist

Push Entitlements check:
The provisioning profile being used for this project appears to have the Push Notification entitlement

Frameworks Search Path check:
Framework Search Paths PASSED

**Comment by elcharlie** [comment]
Thank you

**Comment by DelphiWorlds** [comment]
For the original error, apparently there can be a number of causes. Some things to check for: Make sure the provisioning profile is using the correct certificate. Check which certificates are installed on the Mac by using the Keychain Access app to make sure there's none that might "confuse" PAServer by selecting the wrong one when it attempts to codesign the application. I have tool called [Mosco](https://github.com/DelphiWorlds/Mosco), that can also help identify these kinds of problems - use the View Certs and Profiles menu item to display which certificates and profiles are installed/downloaded.

For the error you're receiving when deploying/running in development mode, are you deploying to a real device? The error apparently can also be affected by misconfiguration of the provisioning profile

**Comment by elcharlie** [comment]
You are right, it was a certificate problem, but it is strange because that same certificate was working without problems for me with another app, which was receiving push notifications in the old way. I don't know, I revoked the certificate, I regenerated it and everything worked fine.
Sorry for the inconvenience, and thank you very much for your help.
But I have a question, with this version of Firebase iOS sdk, on iOS devices, can't I receive an image in the notification?
Many thanks for everything.
All the best.

**Comment by DelphiWorlds** [comment]
> But I have a question, with this version of Firebase iOS sdk, on iOS devices, can't I receive an image in the notification?

I'm working on that, however moving FCM iOS support forward (to newer SDKs) is currently at a standstill due to a linker bug in Delphi, and I'm not inclined to spend any more time on it because it may be completely wasted due to potential restrictions on apps that use older SDKs.

> And another thing that I have detected on Android devices, if you set the FCM.ShowBannerIfForeground property to true, if a notification is sent, of type no data, and the app is running in the foreground, it crashes.

Please file a separate issue for this.

**Comment by elcharlie** [comment]
Yes of course.
I open a new thread.
Thank you so much.
